src/PIResponse.php

```diff
@@ -82,6 +85,20 @@ public static function fromJSON($json, PrivacyIDEA $privacyIDEA)
         {
             $ret->transactionID = $map['detail']['transaction_id'];
         }
+
+        $r = $map['result']['authentication'] ?: null;
```

Comment:
All of this can be replaced with
`
$ret->authenticationStatus = $map['result']['authentication'] ?: AuthenticationStatus::NONE`